api/src/main/java/org/apache/iceberg/Files.java

```diff
@@ -24,8 +24,8 @@
 import java.io.RandomAccessFile;
 import java.nio.file.Paths;
 import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.FileHandlingException;
```

Comment:
Don't we have an overlap between this PR and #6887 ? 
Maybe worth to define an order and gather both PRs in a single PR.

Reply:
Yeah. Description mentions that this PR depends on #6887

Idea is that, I will rebase this PR after #6887 is merged. 
Reviewer can easily review this PR as it will just blind replacement once I rebase.

Reply:
Awesome, thanks :) It will be easier to review 👍